### PR TITLE
New-ParametersFile Addition +semver: minor

### PR DESCRIPTION
**Pull request by adam230594**

### CODE REVIEW

File: Infrastructure-Scripts/New-ParametersFile.ps1 [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    $ParameterObjects = $TemplateParameters.PSObject.Members | Where-Object MemberType -eq NoteProperty
+
+    foreach ($ParameterObject in $ParameterObjects) {
+        $ParameterType = $ParameterObject.Value.Type
+        $ParameterName = $ParameterObject.Name
+        $ParameterValue = (Get-Item -Path "env:$ParameterName" -ErrorAction SilentlyContinue).Value
+
+        if (!$ParameterValue) {
+            Write-Verbose -Message "Environment variable for $ParameterName was not found, attempting default value"
+            $ParameterValue = $TemplateParameters.$ParameterName.defaultValue
```
Comment (smneal) [inline Infrastructure-Scripts/New-ParametersFile.ps1:42]: Does this need to be `$ParameterValue = $ParameterObject.defaultValue` ?

Comment (adam230594) [reply Infrastructure-Scripts/New-ParametersFile.ps1:42]: The $ParameterObject returns an object similar to below: 
```Value           : @{type=string; defaultValue=String}
MemberType      : NoteProperty
IsSettable      : True
IsGettable      : True
TypeNameOfValue : System.Management.Automation.PSCustomObject
Name            : DefaultString
IsInstance      : True
```

The ParameterObjects are used to distinguish the Parameter Name and Parameter Type. 
It is possible to do `$ParameterValue = $ParameterObject.Value.defaultValue` but it goes back into the original arm template and searches for the specific parameter and its default value instead.

Comment (smneal) [reply Infrastructure-Scripts/New-ParametersFile.ps1:42]: OK, I think as you're already referencing the $ParameterObject within the foreach loop I think it makes sense to keep referencing that. Maybe it also has some performance benefit as it doesn't have to search through the entire parameter list again?

Comment (adam230594) [reply Infrastructure-Scripts/New-ParametersFile.ps1]: Yep, does make more sense and good point on the performance benefit as well. Updated 👍

**Pull request merged**